js/utils/dom.js

```diff
@@ -124,26 +106,27 @@ export function handleLinks(el) {
   });
 }
 
-export function openExternal(href) {
-  if (typeof href !== 'string' || !href) {
-    throw new Error('Please provide an href as a string.');
+export function openExternal(link) {
+  if (!link || !link.href || typeof link.href !== 'string') {
+    throw new Error('Please provide a valid link as anchor element with valid href as string.');
   }
+
   const activeServer = app.serverConfigs.activeServer;
   const localSettings = app.localSettings;
   const warningOptedOut = app.localSettings &&
     localSettings.get('dontShowTorExternalLinkWarning');
 
   if (activeServer && activeServer.get('useTor') && !warningOptedOut) {
-    const warningModal = new TorExternalLinkWarning({ url: href })
+    const warningModal = new TorExternalLinkWarning({ url: link.href })
       .render()
       .open();
 
     warningModal.on('cancelClick', () => warningModal.close());
     warningModal.on('confirmClick', () => {
-      shell.openExternal(href);
+      shell.openExternal(link.protocol === 'file:' ? `http://${link.href}` : link.href);
```

Comment:
@jashot7 Could we please keep this method simple and revert back to it just accepting an href?

I think we could put the whole `file:` protocol complexity back in handleLinks() when we make the call, e.g.

```
this.openExternal(link.protocol === 'file:' ? `http://${link.href}` : link.href);
```

openExternal is something I think people should just call with a string url and nothing else.